src/sentry/api/endpoints/debug_files.py

```diff
@@ -177,23 +179,31 @@ def get(self, request: Request, project) -> Response:
         else:
             q = Q()
 
-        file_format_q = Q()
-        for file_format in file_formats:
-            known_file_format = DIF_MIMETYPES.get(file_format)
-            if known_file_format:
-                file_format_q |= Q(file__headers__icontains=known_file_format)
+        if file_formats:
+            file_format_q = Q()
+            for file_format in file_formats:
+                known_file_format = DIF_MIMETYPES.get(file_format)
+                if known_file_format:
+                    file_format_q |= Q(file__headers__icontains=known_file_format)
+            q &= file_format_q
+
+        q &= Q(project_id=project.id)
+        queryset = ProjectDebugFile.objects.filter(q).select_related("file")
 
-        q &= file_format_q
+        def on_results(difs: Sequence[ProjectDebugFile]):
+            # NOTE: we are only refreshing files if there is direct query for specific files
+            if not query and not file_formats:
+                maybe_renew_debug_files(Q(project_id=project.id) & q, difs)
```

Comment:
Doesn't `q` already have `project_id` condition from line 190?

Reply:
:-D indeed, I missed the refactor when I added it to `q`